File: app/src/main/java/com/comfortable/eyes/ProtectModeState.java
package com.comfortable.eyes;

import android.content.Context;
import android.content.SharedPreferences;

public class ProtectModeState {
    private Context mContext;
    private SharedPreferences pref;
    private SharedPreferences.Editor edit;

    public ProtectModeState(Context context) {
        this.mContext = context;
        pref = this.mContext.getSharedPreferences("ProtectModeState", 0);
        edit = pref.edit();
    }

    public void commitState() {
        edit.commit();
    }

    public void enableProtectMode(boolean b) {
        edit.putBoolean("enableNoti", b);
    }

    public void setNotiCount(int m) {
        edit.putInt("notiTime", m);
        edit.putInt("notiCount", m*1);
    }

    public void setNotUsingCount(int m) {
        edit.putInt("notUsingCount", m*1);
    }

    public void setNotiCountValue(int m) {
        edit.putInt("notiCount", m);
    }

    public void setNotUsingCountValue(int m) {
        edit.putInt("notUsingCount", m);
    }

    public void setNotiCountPause(Boolean b) {
        edit.putBoolean("PAUSE_notiCount", b);
    }

    public void setNotUsingCountPause(Boolean b) {
        edit.putBoolean("PAUSE_notUsingCount", b);
    }

    public Boolean isProtectModeEnable() {
        return pref.getBoolean("enableNoti", false);
    }

    public int getNotiCountValue() {
        return pref.getInt("notiCount",1);
    }

    public int getNotUsingCountValue() {
        return pref.getInt("notUsingCount", 1);
    }

    public int getNotiTime() {
        return pref.getInt("notiTime", 15);
    }

    public Boolean isNotiCountPaused() {
        return pref.getBoolean("PAUSE_notiCount", false);
    }

    public Boolean isNotUsingCountPaused() {
        return pref.getBoolean("PAUSE_notUsingCount", false);
    }
}

File: app/src/main/java/com/comfortable/eyes/state/ProtectModeState.kt
package com.comfortable.eyes

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor

class ProtectModeState(private val mContext: Context) {
    private val pref: SharedPreferences
    private val edit: Editor
    fun commitState() {
        edit.commit()
    }

    fun enableProtectMode(b: Boolean) {
        edit.putBoolean("enableNoti", b)
    }

    fun setNotiCount(m: Int) {
        edit.putInt("notiTime", m)
        edit.putInt("notiCount", m * 60)
    }

    fun setNotUsingCount(m: Int) {
        edit.putInt("notUsingCount", m * 60)
    }

    fun setNotiCountPause(b: Boolean?) {
        edit.putBoolean("PAUSE_notiCount", b!!)
    }

    fun setNotUsingCountPause(b: Boolean?) {
        edit.putBoolean("PAUSE_notUsingCount", b!!)
    }

    val isProtectModeEnable: Boolean
        get() = pref.getBoolean("enableNoti", false)
    var notiCountValue: Int
        get() = pref.getInt("notiCount", 1)
        set(m) {
            edit.putInt("notiCount", m)
        }
    var notUsingCountValue: Int
        get() = pref.getInt("notUsingCount", 1)
        set(m) {
            edit.putInt("notUsingCount", m)
        }
    val notiTime: Int
        get() = pref.getInt("notiTime", 15)
    val isNotiCountPaused: Boolean
        get() = pref.getBoolean("PAUSE_notiCount", false)
    val isNotUsingCountPaused: Boolean
        get() = pref.getBoolean("PAUSE_notUsingCount", false)

    init {
        pref = mContext.getSharedPreferences("ProtectModeState", 0)
        edit = pref.edit()
    }
}